### ISSUE TITLE
NCE loss function?

### ISSUE THREAD

**Issue by yanxp** [akwasigroch/Pretext-Invariant-Representations#1]
https://github.com/akwasigroch/Pretext-Invariant-Representations/blob/c2b4d65d96e89ba67b20f5831808d9f67451a7af/utils.py#L288
Hello, why the line is different from the paper to compute the nce loss? Thanks.

**Comment by akwasigroch** [comment]
Note, that h function (equation 3 from the paper) can be interpreted as softmax function and similarities between images can be interpreted as logits. Therefore, I used torch.nn.CrossEntropyLoss() because this loss whit this interpretation is equivalent to the loss described by equation 4.

**Comment by Chaimmoon** [comment]
Yes, your first part of Equ 4 is right. However, the second part 
![image](https://user-images.githubusercontent.com/41274366/78564115-8efdc780-784e-11ea-8441-97faf06a68a1.png) is not included in your code, which is different from the original CE loss.

**Comment by akwasigroch** [comment]
I may not be correct, but I think that minimization of the first term of Equ 4 implicitly minimizes the second term - as in the softmax layer - the maximization of probability of one class leads to minimization of probabilities of other classess.

**Comment by Crane-YU** [comment]
Sorry for reopening this issue again. I think the loss function from Eq. 4 in the paper can be implemented as a binary cross entropy loss. I agree with @akwasigroch that `h()` in Eq. 3 is actually the `softmax()`, and the second part mentioned by @Chaimmoon can be computed as well if using `torch.nn.BCEWithLogitsLoss()`
